runtime/vm/MHInterpreter.cpp

```diff
@@ -555,6 +555,33 @@ VM_MHInterpreter::dispatchLoop(j9object_t methodHandle)
 			/* Get MethodHandle for this operation from the VarHandle's handleTable */
 			j9object_t handleTable = J9VMJAVALANGINVOKEVARHANDLE_HANDLETABLE(_currentThread, varHandle);
 			j9object_t methodHandleFromTable = J9JAVAARRAYOFOBJECT_LOAD(_currentThread, handleTable, operation);
+
+			if (NULL == methodHandleFromTable) {
+				/* Building a method type (MT) frame makes the stack walkable since looking up the class below may
+				 * cause a GC or another exception. Also, the MT frame will make it easier to debug and service the
+				 * error as the VarHandle will be on the stack and findable with DDR. The MT frame does not need to
+				 * be restored since throwing the exception will handle it appropriately.
+				 */
+				buildMethodTypeFrame(_currentThread, type);
+
+				const char *exceptionClassName = "java/lang/UnsupportedOperationException";
+				J9Class *exceptionClass = internalFindClassUTF8(
+						_currentThread,
+						(U_8 *)exceptionClassName,
+						strlen(exceptionClassName),
+						_currentThread->javaVM->systemClassLoader,
+						J9_FINDCLASS_FLAG_THROW_ON_FAIL);
+
+				j9object_t exception = _objectAllocate->inlineAllocateObject(_currentThread, exceptionClass);
```

Comment:
Same issue here - the inlineAllocateObject only indicates if we can allocate without a GC or not.  When it returns null, we usually need to build a frame / make the stack walkable & then call the gc'ing version of the allocate helpers